src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -316,6 +316,38 @@ public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
     return success;
   }
 
+  @Override
+  public CourseConfirmState getConfirmeState(DocumentReference objDocRef) {
+    DocumentReference courseParticipantClassRef = getCourseClasses().getCourseParticipantClassRef(
+        modelContext.getXWikiContext().getDatabase());
+    CourseConfirmState confirmState = CourseConfirmState.UNCONFIRMED;
+    try {
+      List<BaseObject> partiObjs = modelAccess.getXObjects(objDocRef, courseParticipantClassRef);
+      int index = 0;
+      for (BaseObject obj : partiObjs) {
+        String state = obj.getStringValue("status");
```

Comment:
please use use `CourseConfirmState.valueOf(Strings.nullToEmpty(obj.getStringValue("status")))` to convert to a CourseConfirmState